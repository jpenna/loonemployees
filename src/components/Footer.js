import React from 'react';

export default function Footer() {
  return (
    <footer className="footer hero is-link">
      <p className="content has-text-centered">
        Thats all Folks!
      </p>
    </footer>
  );
}
