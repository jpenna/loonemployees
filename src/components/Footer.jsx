import React from 'react';

export default function Footer() {
  return (
    <footer>
      <p>
        Thats all Folks!
      </p>
    </footer>
  );
}
