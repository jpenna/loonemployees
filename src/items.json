[{
  "id": 1,
  "avatar": "https://imgc.allpostersimages.com/img/print/poster/bugs-bunny_a-G-10260056-0.jpg",
  "name": "Bugs Bunny",
  "bio": "Bugs Bunny is an animated cartoon character, created in the late 1930s by Leon Schlesinger Productions and voiced originally by Mel Blanc. Bugs is best known for his starring roles in the Looney Tunes and Merrie Melodies series of animated short films, produced by Warner Bros."
}, {
  "id": 4,
  "name": "Tasmanian Devil",
  "bio": "The Tasmanian Devil, commonly referred to as Taz, is an animated cartoon character featured in the Warner Bros. Looney Tunes and Merrie Melodies series of cartoons. The Tasmanian Devil, commonly referred to as Taz, is an animated cartoon character featured in the Warner Bros. Looney Tunes and Merrie Melodies series of cartoons."
}, {
  "id": 2,
  "avatar": "https://vignette.wikia.nocookie.net/parody/images/2/2e/Daffy-duck.png/revision/latest?cb=20180523005334",
  "name": "Daffy Duck",
  "bio": "Daffy Duck is an animated cartoon character produced by Warner Bros. Styled as an anthropomorphic black duck, the character has appeared in cartoon series such as Looney Tunes and Merrie Melodies, in which he usually has been depicted as a foil for Bugs Bunny."

}, {
  "id": 3,
  "avatar": "https://vignette.wikia.nocookie.net/looneytunes/images/e/e7/Tweety1.gif/revision/latest?cb=20140406233615",
  "name": "Tweety",
  "bio": "Tweety is an animated fictional yellow canary in the Warner Bros. Looney Tunes and Merrie Melodies series of animated cartoons. The name \"Tweety\" is a play on words, as it originally meant \"sweetie\", along with \"tweet\" being a typical English chick for the sounds of birds."
}]
