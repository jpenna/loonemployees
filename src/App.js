import React, { Component } from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router-dom';

import EmployeeList from './components/EmployeeList';
import NotFound from './components/NotFound';
import Footer from './components/Footer';

import itemsJSON from './items.json';

export class AppComponent extends Component {
  constructor(props) {
    super(props);
    this.state = { items: itemsJSON };

    this.onClickBio = this.onClickBio.bind(this);
  }

  onClickBio({ e, id }) {
    e.stopPropagation();
    e.currentTarget.blur(); // Remove focus outline on click, but keep for accessibility
    const randomHsl = `hsla(${Math.random() * 360}, 60%, 80%, 0.4)`;

    this.setState((prevState) => {
      const { items: prevItems } = prevState;
      const items = [...prevItems];
      const index = items.findIndex(i => i.id === id);
      const item = { ...items[index], backgroundColor: randomHsl };
      items[index] = item;
      return { items };
    });
  }

  render() {
    const { items } = this.state;

    return (
      <div className="app-container">
        <div className="container app-content">
          <Switch>
            <Route
              exact
              path="/"
              render={props => (
                <EmployeeList
                  {...props}
                  items={items}
                  onClickBio={this.onClickBio}
                />
              )}
            />
            <Route component={NotFound} />
          </Switch>
        </div>
        <Footer />
      </div>
    );
  }
}

export default hot(module)(AppComponent);
