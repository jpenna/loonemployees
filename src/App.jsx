import React, { Component } from 'react';
import { hot } from 'react-hot-loader';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import EmployeeList from './components/EmployeeList';
import NotFound from './components/NotFound';
import Footer from './components/Footer';

import itemsJSON from './items.json';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = { items: itemsJSON };
  }

  render() {
    const { items } = this.state;

    return (
      <Router>
        <div className="app-container">
          <div className="container app-content">
            <Switch>
              <Route exact path="/" render={props => <EmployeeList {...props} items={items} />} />
              <Route component={NotFound} />
            </Switch>
          </div>
          <Footer />
        </div>
      </Router>
    );
  }
}

export default hot(module)(App);
