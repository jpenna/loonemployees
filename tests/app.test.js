/* eslint import/no-extraneous-dependencies: "off" */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { AppComponent } from '../src/App';
import Footer from '../src/components/Footer';

describe('<App />', () => {
  let shallowed;

  beforeAll(() => {
    shallowed = shallow(<AppComponent />);
  });

  test('should have <Footer /> component', () => {
    expect(shallowed.contains(<Footer />)).toBeTruthy();
  });

  test('should render correctly', () => {
    const rendered = renderer.create(<AppComponent />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
