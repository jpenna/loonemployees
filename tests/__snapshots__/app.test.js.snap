// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<div
  className="app-container"
>
  <div
    className="container app-content"
  >
    <div
      className="section"
    >
      <div
        className="title is-3 has-text-centered"
      >
        Employees List
      </div>
      <div
        className="box employee-item clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
        tabIndex="0"
      >
        <div
          className="media "
        >
          <figure
            className="media-left"
          >
            <p
              className="image is-64x64"
            >
              <img
                alt="Bugs Bunny"
                src="https://imgc.allpostersimages.com/img/print/poster/bugs-bunny_a-G-10260056-0.jpg"
              />
            </p>
          </figure>
          <div
            className="media-content "
          >
            <div
              className="content"
            >
              <h3
                className="title is-5 mb-1"
              >
                Bugs Bunny
              </h3>
              <p
                className="text-ellipsis"
              >
                Bugs Bunny is an animated cartoon character, created in the late 1930s by Leon Schlesinger Productions and voiced originally by Mel Blanc. Bugs is best known for his starring roles in the Looney Tunes and Merrie Melodies series of animated short films, produced by Warner Bros.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="box employee-item clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
        tabIndex="0"
      >
        <div
          className="media "
        >
          <div
            className="media-content expand"
          >
            <div
              className="content"
            >
              <h3
                className="title is-5 mb-1"
              >
                Tasmanian Devil
              </h3>
              <p
                className="text-ellipsis"
              >
                The Tasmanian Devil, commonly referred to as Taz, is an animated cartoon character featured in the Warner Bros. Looney Tunes and Merrie Melodies series of cartoons. The Tasmanian Devil, commonly referred to as Taz, is an animated cartoon character featured in the Warner Bros. Looney Tunes and Merrie Melodies series of cartoons.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="box employee-item clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
        tabIndex="0"
      >
        <div
          className="media "
        >
          <figure
            className="media-left"
          >
            <p
              className="image is-64x64"
            >
              <img
                alt="Daffy Duck"
                src="https://vignette.wikia.nocookie.net/parody/images/2/2e/Daffy-duck.png/revision/latest?cb=20180523005334"
              />
            </p>
          </figure>
          <div
            className="media-content "
          >
            <div
              className="content"
            >
              <h3
                className="title is-5 mb-1"
              >
                Daffy Duck
              </h3>
              <p
                className="text-ellipsis"
              >
                Daffy Duck is an animated cartoon character produced by Warner Bros. Styled as an anthropomorphic black duck, the character has appeared in cartoon series such as Looney Tunes and Merrie Melodies, in which he usually has been depicted as a foil for Bugs Bunny.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="box employee-item clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
        tabIndex="0"
      >
        <div
          className="media "
        >
          <figure
            className="media-left"
          >
            <p
              className="image is-64x64"
            >
              <img
                alt="Tweety"
                src="https://vignette.wikia.nocookie.net/looneytunes/images/e/e7/Tweety1.gif/revision/latest?cb=20140406233615"
              />
            </p>
          </figure>
          <div
            className="media-content "
          >
            <div
              className="content"
            >
              <h3
                className="title is-5 mb-1"
              >
                Tweety
              </h3>
              <p
                className="text-ellipsis"
              >
                Tweety is an animated fictional yellow canary in the Warner Bros. Looney Tunes and Merrie Melodies series of animated cartoons. The name "Tweety" is a play on words, as it originally meant "sweetie", along with "tweet" being a typical English chick for the sounds of birds.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer hero is-link"
  >
    <p
      className="content has-text-centered"
    >
      Thats all Folks!
    </p>
  </footer>
</div>
`;
